cmd/daemonset-check/run_check.go

```diff
@@ -337,6 +337,14 @@ func generateDaemonSetSpec() *appsv1.DaemonSet {
 		log.Errorln("Error getting ownerReference:", err)
 	}
 
+	ownerPod, err := client.CoreV1().Pods(checkNamespace).Get(context.Background(), ownerRef[0].Name, metav1.GetOptions{})
```

Comment:
Can you add a check above for `len(ownerRef) > 0` please?

Reply:
sure